src/schroedinger.jl

```diff
@@ -102,3 +104,18 @@ function check_schroedinger(psi::Bra, H)
     check_multiplicable(psi, H)
     check_samebases(H)
 end
+
+
+_promote_time_and_state(tspan, psi0, f) = _promote_time_and_state(tspan, psi0, f(first(tspan), psi0))
+function _promote_time_and_state(tspan, psi0, H::AbstractOperator)
+    # general case is Ts<:Complex, Tt<:Real
+    Ts = eltype(H)
+    Tt = real(Ts)
+    (isconcretetype(Ts) && isconcretetype(Tt)) || @warn "For using `ForwardDiff` on `schroedinger` the element type of `real(H(t,psi)*psi)` must be concrete !!\nGot elements of type $Tt \nTry promoting the Hamiltonian elements based on the parameters you are differentiating by."
+    tspan = Tt.(tspan)
```

Comment:
I imagine there are situations where no promotion is necessary. Can we shortcircuit those?

Reply:
I'll declare type `Dual` and only promote that case, probably would be safer.